utils/pytorch.py

```diff
@@ -35,10 +37,20 @@ def __init__(self, model, disable_jit_freeze=False, example_inputs=None):
             if AIO:
                 utils.print_warning_message(f"Running with disable_jit_freeze={disable_jit_freeze} - Ampere optimizations are not expected to work.")
         else:
-            try:
-                self.__frozen_script = torch.jit.freeze(torch.jit.script(self.__model))
-            except torch.jit.frontend.UnsupportedNodeError:
-                self.__frozen_script = torch.jit.freeze(torch.jit.trace(self.__model, example_inputs))
+            cached_dir = Path(__file__).resolve().parents[1] / "cached"
```

Comment:
Using the solution here for this case: https://github.com/AmpereComputingAI/model_zoo_dev/blob/main/downloads/utils.py
seems more clear to me.